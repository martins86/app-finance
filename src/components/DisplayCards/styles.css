.display-cards {
    max-width: 1120px;
    width: 92%;
    margin: 0 auto;
    display: flex;
    margin-top: -50px;
    justify-content: space-between;
}