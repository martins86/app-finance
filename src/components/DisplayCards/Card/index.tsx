import './styles.css';

export interface CardParams {
    titleCard?: string;
    Icon?: any;
    iconColor?: string;
    value?: number;
}

const Card = ({ titleCard, Icon, iconColor, value }: CardParams) => {
    return (
        <div className="card">
            <header className="card-header">
                <h4 className="card-title">{titleCard}</h4>
                <Icon color={iconColor} />
            </header>
            <h5 className="total">
                {value?.toLocaleString('pt-br',{style: 'currency', currency: 'BRL'})}
            </h5>
        </div>
    );
};

export default Card;