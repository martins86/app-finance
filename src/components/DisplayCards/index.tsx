import Card from './Card';

import {
  FaRegArrowAltCircleUp,
  FaRegArrowAltCircleDown,
  FaDollarSign,
} from 'react-icons/fa';

import './styles.css';

export interface DisplayCardsParams {
  income?: number;
  expense?: number;
  total?: number;
}

const DisplayCards = ({ income, expense, total }: DisplayCardsParams) => {
  return (
    <div className="display-cards">
      <Card
        titleCard="Entradas"
        Icon={FaRegArrowAltCircleUp}
        iconColor="royalblue"
        value={income}
      />
      <Card
        titleCard="SaÃ­das"
        Icon={FaRegArrowAltCircleDown}
        iconColor="red"
        value={expense}
      />
      <Card
        titleCard="Total"
        Icon={FaDollarSign}
        iconColor="green"
        value={total}
      />
    </div>
  );
};

export default DisplayCards;
