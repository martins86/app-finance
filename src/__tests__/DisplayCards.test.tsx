import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import DisplayCards from '../components/DisplayCards';

describe('Test <DisplayCards />', () => {
  test('Should match snapshot', () => {
    // Arrange
    const { container } = render(
      <Router>
        <DisplayCards />
      </Router>
    );

    // Act & Assert
    expect(container).toMatchSnapshot();
  });
});
