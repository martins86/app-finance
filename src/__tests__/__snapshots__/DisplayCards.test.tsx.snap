// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <DisplayCards /> Should match snapshot 1`] = `
<div>
  <div
    class="display-cards"
    data-testid="display-cards"
  >
    <div
      class="card"
    >
      <header
        class="card-header"
      >
        <h4
          class="card-title"
        >
          Entradas
        </h4>
        <svg
          color="royalblue"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="color: royalblue;"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 504c137 0 248-111 248-248S393 8 256 8 8 119 8 256s111 248 248 248zm0-448c110.5 0 200 89.5 200 200s-89.5 200-200 200S56 366.5 56 256 145.5 56 256 56zm20 328h-40c-6.6 0-12-5.4-12-12V256h-67c-10.7 0-16-12.9-8.5-20.5l99-99c4.7-4.7 12.3-4.7 17 0l99 99c7.6 7.6 2.2 20.5-8.5 20.5h-67v116c0 6.6-5.4 12-12 12z"
          />
        </svg>
      </header>
      <h5
        class="total"
      />
    </div>
    <div
      class="card"
    >
      <header
        class="card-header"
      >
        <h4
          class="card-title"
        >
          Sa√≠das
        </h4>
        <svg
          color="red"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="color: red;"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200zm-32-316v116h-67c-10.7 0-16 12.9-8.5 20.5l99 99c4.7 4.7 12.3 4.7 17 0l99-99c7.6-7.6 2.2-20.5-8.5-20.5h-67V140c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12z"
          />
        </svg>
      </header>
      <h5
        class="total"
      />
    </div>
    <div
      class="card"
    >
      <header
        class="card-header"
      >
        <h4
          class="card-title"
        >
          Total
        </h4>
        <svg
          color="green"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="color: green;"
          viewBox="0 0 288 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M209.2 233.4l-108-31.6C88.7 198.2 80 186.5 80 173.5c0-16.3 13.2-29.5 29.5-29.5h66.3c12.2 0 24.2 3.7 34.2 10.5 6.1 4.1 14.3 3.1 19.5-2l34.8-34c7.1-6.9 6.1-18.4-1.8-24.5C238 74.8 207.4 64.1 176 64V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48h-2.5C45.8 64-5.4 118.7.5 183.6c4.2 46.1 39.4 83.6 83.8 96.6l102.5 30c12.5 3.7 21.2 15.3 21.2 28.3 0 16.3-13.2 29.5-29.5 29.5h-66.3C100 368 88 364.3 78 357.5c-6.1-4.1-14.3-3.1-19.5 2l-34.8 34c-7.1 6.9-6.1 18.4 1.8 24.5 24.5 19.2 55.1 29.9 86.5 30v48c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-48.2c46.6-.9 90.3-28.6 105.7-72.7 21.5-61.6-14.6-124.8-72.5-141.7z"
          />
        </svg>
      </header>
      <h5
        class="total"
      />
    </div>
  </div>
</div>
`;
